libs/sql-schema-describer/src/postgres.rs

```diff
@@ -214,16 +205,64 @@ impl SqlSchemaDescriber {
     }
 
     fn get_precision(col: &ResultRow) -> Precision {
-        let character_maximum_length = col.get_u32("character_maximum_length");
-        let numeric_precision = col.get_u32("numeric_precision");
-        let numeric_precision_radix = col.get_u32("numeric_precision_radix");
-        let numeric_scale = col.get_u32("numeric_scale");
-        let time_precision = col.get_u32("datetime_precision");
+        let (character_maximum_length, numeric_precision, numeric_scale, time_precision) =
+            if matches!(col.get_expect_string("data_type").as_str(), "ARRAY") {
+                fn get_single(formatted_type: &String) -> Option<u32> {
+                    static SINGLE_REGEX: Lazy<Regex> = Lazy::new(|| Regex::new(r#".*\(([0-9]*)\).*\[\]$"#).unwrap());
+
+                    SINGLE_REGEX
+                        .captures(formatted_type.as_str())
+                        .and_then(|cap| cap.get(1).map(|precision| from_str::<u32>(precision.as_str()).unwrap()))
+                }
+
+                fn get_dual(formatted_type: &String) -> (Option<u32>, Option<u32>) {
+                    static DUAL_REGEX: Lazy<Regex> =
+                        Lazy::new(|| Regex::new(r#"numeric\(([0-9]*),([0-9]*)\)\[\]$"#).unwrap());
+                    let first = DUAL_REGEX
+                        .captures(formatted_type.as_str())
+                        .and_then(|cap| cap.get(1).map(|precision| from_str::<u32>(precision.as_str()).unwrap()));
+
+                    let second = DUAL_REGEX
+                        .captures(formatted_type.as_str())
+                        .and_then(|cap| cap.get(2).map(|precision| from_str::<u32>(precision.as_str()).unwrap()));
+
+                    (first, second)
+                }
+
+                let formatted_type = col.get_expect_string("formatted_type");
+                let fdt = col.get_expect_string("full_data_type");
+                let char_max_length = if matches!(fdt.as_str(), "_bpchar" | "_varchar" | "_bit" | "_varbit") {
```

Comment:
If having already if/else, using `match` instead of `matches!` would look cleaner.

Reply:
agree